### ISSUE TITLE
"Whoops" error page should be more helpful

### ISSUE THREAD

**Issue by milescalabresi** [learning-unlimited/ESP-Website#3542]
There are a handful of issues that propose showing more specific error pages for certain errors. More broadly, I think we should move away from the generic "whoops" page and towards a page that describes what the error is, what (if anything) an admin can do about it, and what (if anything) to say to the web team member.

**Comment by willgearty** [comment]
Keep in mind that often these error pages are seen by non-admin users, so we don't want any private information or information that could be used for hacking, etc to be shown on the page. Further, it should be clear what steps normal users could do to solve the problem (if any) vs. what steps admins could do.

**Comment by milescalabresi** [comment]
That is true. Maybe we should put an `if` tag in the template and give more information if the user is logged in as an admin

**Comment by Supriyo760** [comment]
Hi @milescalabresi and @willgearty, I'd love to work on this issue! Here's what I'm thinking:

What I found in the codebase: There are currently three separate error templates (error.html, 500.html, 403.html) that are all slightly different. There's even a TODO comment in esp/error/middleware.py saying they should be merged -so I'd tackle that too.

My plan:

1. Merge the error templates into one shared base template to reduce duplication.

2. Two levels of detail (as you both discussed):
   - Regular users see: a friendly message, suggested next steps, and the support email.
   - Admins see: the technical error details, exception type, and Sentry ID -gated behind {% if request.user.isAdministrator %}.

3. Error-specific messages so each type of error (403, 500, ESPError) shows a different helpful description instead of a generic "Whoops."

4. One-click "Report this Error" button -when clicked, it opens the user's email app with everything already pre-filled: the page URL, timestamp, and Sentry error ID in the body. The user just adds one line describing what they were doing and hits send. No more manually copying error IDs or writing the email from scratch. This directly solves the "what to say to the web team" part of the issue. For admins, the pre-filled email also includes technical details like the exception type.

Let me know if this sounds good and I'll get started!

**Comment by willgearty** [comment]
@Supriyo760 that sounds like a great plan, thanks!

**Comment by willgearty** [comment]
@Supriyo760 Thanks for your interest in contributing to our project! I've assigned you to this issue. Please make sure to check out our [instructions for getting started](https://github.com/learning-unlimited/ESP-Website/wiki#getting-started) and follow our [contribution rules](https://github.com/learning-unlimited/ESP-Website/wiki#rules). Let me know if you have any questions.

**Comment by Supriyo760** [comment]
@willgearty I've completed the PR and it's ready for review whenever you have time.

**Comment by Supriyo760** [comment]
Hi @willgearty! I've pushed a new commit to PR #4396 that addresses all of the feedback generated by Copilot. I added the necessary ARIA labels for accessibility, fixed the 500.html context variables in web.py, ensured 403 errors still show specific permission messages, and resolved the other edge cases it pointed out. Whenever you have a moment to take a look, I'd appreciate a quick review!

**Comment by willgearty** [comment]
Thanks 👍 No need to tag me for future commits, though. And please keep all messages in the PR.
